ext_mongodb.php

```diff
@@ -19,6 +19,110 @@ function bsonUnserialize(array $data) : void;
 {
 }
 
+/* == MONITORING ============================================================= */
+namespace MongoDB\Driver\Monitoring;
+
+/* ------ Building blocks for value obejcts */
+abstract class _CommandEvent
+{
+	private string                 $commandName;
+	private \MongoDB\Driver\Server $server;
+	private integer                $operationId;
+	private integer                $requestId;
+
+	private function __construct()
+	{
+		throw new \MongoDB\Driver\Exception\RunTimeException("Accessing private constructor");
+	}
+
+	public function getCommandName() : string
+	{
+		return $this->commandName;
+	}
+
+	// libmongoc use getHost, and the  spec uses "connectionId".
+	public function getServer() : \MongoDB\Driver\Server
+	{
+		return $this->server;
+	}
+
+	// technically int64_t
+	public function getOperationId() : string
+	{
+		return (string) $this->operationId;
+	}
+
+	// technically int64_t
+	public function getRequestId() : string
+	{
+		return (string) $this->requestId;
+	}
+}
+
+abstract class _CommandResultEvent extends _CommandEvent
+{
+	private integer $durationMicros;
+
+	// spec is unitless, libmongoc does microseconds. Spec suggests "Nanos" as
+	// another alternative. (technically int64_t, but 2147 sec on 32bit platforms
+	// ought to be enough)
+	public function getDurationMicros() : integer
+	{
+		return $this->durationMicros;
+	}
+}
+
+/* ------ Value Objects ------------------- */
+final class CommandStartedEvent extends _CommandEvent
+{
+	private object $command;
+	private string $databaseName;
+
+	// raw command document, with default BSON conversion rules
+	public function getCommand() : object
```

Comment:
In anticipation of supporting type maps in the future without breaking BC, should we relax this return type to `mixed` (I assume `array | object` is not allowed)?